lib/techschool_web/live/course_live/index.html.heex

```diff
@@ -9,11 +9,20 @@
     selected_tool={@selected_tool}
   />
   <div class="text-center">
-    <div class="flex flex-wrap justify-center gap-8 md:gap-10" id="courses" phx-update="stream">
-      <.course_card :for={{dom_id, course} <- @streams.courses} id={dom_id} course={course} />
-    </div>
-    <.button phx-click="load_more" class="mt-8" disabled={@has_more_courses_to_load == false}>
-      <%= gettext("Load More") %>
-    </.button>
+    <%= if @has_courses do %>
+      <div class="flex flex-wrap justify-center gap-8 md:gap-10" id="courses" phx-update="stream">
+        <.course_card :for={{dom_id, course} <- @streams.courses} id={dom_id} course={course} />
+      </div>
+    <% else %>
+      <div class="flex flex-wrap justify-center gap-8 md:gap-10">
+        <p>No results for the given search</p>
+      </div>
+    <% end %>
+
+    <%= if @has_courses and @has_more_courses_to_load do %>
+      <.button phx-click="load_more" class="mt-8" disabled={@has_more_courses_to_load == false}>
+        <%= gettext("Load More") %>
+      </.button>
+    <% end %>
```

Comment:
I think we can be less verbose, and use the `:if` inside the component. Also, feel free to remove the `disabled`

```suggestion
    <.button phx-click="load_more" class="mt-8" :if={@has_courses and @has_more_courses_to_load}>
      <%= gettext("Load More") %>
    </.button>
```

Reply:
Nice! I work on a elixir/phoenix project that is not updated to the new version, so these :if/:for attributes are totally new to me, thanks for the advice!